docs/azure/sdk/protocol-convenience-methods.md

```diff
@@ -0,0 +1,209 @@
+ChatClient chatClient = client.GetChatClient("gpt-4");
+
+// Create the request prompt content
+BinaryData prompt = BinaryData.FromBytes("""
+    {  
+        "model": "gpt-4o",
+        "messages": [
+           {
+               "role": "user",
+               "content": "What is Microsoft Azure?."
+           }
+        ]
+    }
+    """u8.ToArray());
+using BinaryContent content = BinaryContent.Create(prompt);
+
+// Send the request
+ClientResult result = chatClient.CompleteChat(
+        content,
+        new RequestOptions()
+        { 
+            ErrorOptions = ClientErrorBehaviors.NoThrow
+        });
+
+// Retrieve and display the response data
+BinaryData output = result.GetRawResponse().Content;
+using JsonDocument outputAsJson = JsonDocument.Parse(output);
+string message = outputAsJson.RootElement
+    .GetProperty("choices"u8)[0]
+    .GetProperty("message"u8)
+    .GetProperty("content"u8)
+    .GetString();
+
+Console.WriteLine(message);
+```
+
+The preceding code demonstrates the following `System.ClientModel` protocol method patterns:
+
+- Uses the `BinaryContent` type as a parameter to supply data for the request body.
+- Uses the `RequestContext` type to configure request options.
+- Returns data using the `ClientResult` type.
+- Calls the `GetRawResponse` method to access the response data.
+
+#### Work with convenience methods
+
+Consider the following code that uses a `ChatClient` to call the `CompleteChat` convenience method:
+
+```csharp
+// Create the client
+OpenAIClient client = new("your-openai-key");
+ChatClient chatClient = client.GetChatClient("gpt-4");
+
+// Send the prompt
+ClientResult<ChatCompletion> completion
+    = chatClient.CompleteChat("What is Azure?");
+
+// Display the response data
+Console.WriteLine($"{completion.Value.Role}: {completion.Value.Content}");
+```
+
+The preceding code demonstrates the following `System.ClientModel` convenience method patterns:
+
+- Uses a standard C# primitive or model type as a parameter.
+- Returns a `ClientResult` type that represents the result of the operation.
+
+## Protocol and convenience method usage guidance
+
+Although the Azure SDK for .NET client libraries provide the option to use either protocol or convenience methods, in most scenarios developers should prioritize using convenience methods. Convenience methods are designed to improve the development experience and provide flexibility for authoring requests and handling responses. Consider the following criteria when deciding which type of method to use:
+
+- Convenience methods:
+  - Enable you to work with more friendly method parameter and response types.
+  - Handle various low-level concerns and optimizations for you.
+
+- Protocol methods:
+  - Provide access to lower-level types such as `RequestContext` and `RequestOptions` that are not available through convenience methods.
+  - Enable access to features of the underlying REST APIs that convenience methods do not expose.
```

Comment:
I would qualify that they may in some cases do this ... and it may also be worth adding a third point, that they may also provide service methods corresponding to service endpoints that don't have convenience methods for them yet -- but that those convenience methods may be added in later versions of the library.  Although maybe those are both part of the same point?